source/constructs/lib/agent/rename-resources/rename_resources.py

```diff
@@ -0,0 +1,185 @@
+    response_body['Status'] = response_status
+    response_body['PhysicalResourceId'] = event['PhysicalResourceId'] if 'PhysicalResourceId' in event else event['LogicalResourceId']
+    response_body['StackId'] = event['StackId']
+    response_body['RequestId'] = event['RequestId']
+    response_body['LogicalResourceId'] = event['LogicalResourceId']
+    response_body['Reason'] = reason
+
+    json_response_body = json.dumps(response_body)
+
+    headers = {
+        'content-type': '',
+        'content-length': str(len(json_response_body))
+    }
+    response = requests.put(response_url,
+                            data=json_response_body,
+                            headers=headers)
+    return response
+
+
+def on_create(event):
+    logger.info("Got create")
+
+
+def on_update(event):
+    logger.info("Got Update")
+    list_crawlers()
+
+
+def on_delete(event):
+    logger.info("Got Delete")
+
+
+def rename_database(new_database_name):
+    try:
+        glue_client.get_database(Name=new_database_name)
+        logger.info(f"The new database({new_database_name}) exists and does not need to be created")
+    except glue_client.exceptions.EntityNotFoundException as e:
+        logger.info(f"The new database({new_database_name}) does not exist and needs to be created")
+        response = glue_client.create_database(DatabaseInput={'Name': new_database_name})
+        # wait for database creation, several seconds
+        crawler_role_arn = f'arn:{partition}:iam::{account_id}:role/{solution_name}GlueDetectionJobRole-{region}'
+
+        # retry for grant permissions
+        num_retries = GRANT_PERMISSIONS_RETRIES
+        while num_retries > 0:
+            try:
+                lakeformation_client.grant_permissions(
+                    Principal={
+                        'DataLakePrincipalIdentifier': f"{crawler_role_arn}"
+                    },
+                    Resource={
+                        'Database': {
+                            'Name': new_database_name
+                        }
+                    },
+                    Permissions=['ALL'],
+                    PermissionsWithGrantOption=['ALL']
+                )
+            except Exception as e:
+                time.sleep(SLEEP_MIN_TIME)
+                num_retries -= 1
+            else:
+                break
+
+
+def rename_crawler(old_crawler_name, new_crawler_name, new_database_name, new_connection_name):
+    response = glue_client.get_crawler(Name=old_crawler_name)
+    logger.info(response)
+    old_crawler = response["Crawler"]
+    try:
+        glue_client.get_crawler(Name=new_crawler_name)
+        logger.info(f"The new crawler({new_crawler_name}) exists and does not need to be created")
+    except glue_client.exceptions.EntityNotFoundException as e:
+        logger.info(f"The new crawler({new_crawler_name}) does not exist and needs to be created")
+        jdbc_targets = old_crawler["Targets"]["JdbcTargets"]
+        if len(jdbc_targets) > 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)